### ISSUE TITLE
`xreg` issue when having user-defined forecast function

### ISSUE THREAD

**Issue by earowang** [earowang/hts#28]
I was trying to use forecastHybrid with hierarchical time series and external regressors but I cannot figure out how to do that. I am trying in this way:

```r
library(hts)
library(forecastHybrid )

forecast(htseg1, a.args = list(xreg = data.frame(1:10)), models = "aet", 
xreg = data.frame(11:20), FUN = hybridModel)
```
But I got the following error:

```r
Error in forecast.Arima(object$auto.arima, h = h, xreg = xreg, level = level) : No regressors provided
```

**Comment by earowang** [comment]
If users define their own forecast function where there's an `xreg` argument, then `newxreg` needs to be supplied and should appear in `line 117` of `forecast.gts.R`.

**Comment by nnarukulla** [comment]
I tried hts + hybridmodel. Would like to share my findings on this. I stopped working on this further due to  below requirements.

1.  Computation time:  Taking more time for execution as it needs as high computation power. 
 In my case, i tried  with more than 2000 bottom level series( total all level series will be 3000+). Due to time limitations, I run  hts for each model like fmethod=arima, FUN=nnetar... independently. Then gave equal weights by averaging forecast values. This satisfies only equal weights hybridmodel and a lot quicker than using hybridmodel with equal weights.

2. Calling different xreg for each series, currently hts doesnt alllow this. So, going with combinef format.

3. CV for weights. CV requires more computation and inside hybridmodel, CV is limited to static window size. Since, 60% of my bottom level series are having less than 2 years data, application of CV getting difficult. Instead I just started exploring with tsCV in latest forecast package. Hopefully this works better.
 
Hope this helps your work on hts package.
